engine/scene.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "colliderbox.h"
+#include "gameobject.h"
+#include "player.h"
+#include "rectangle.h"
+#include "transform.h"
+
+#include <string>
+#include <memory>
+#include <vector>
+
+namespace pong
+{
+
+using GameObjectCollection = std::vector<std::unique_ptr<GameObject>>;
+
+class Scene
+{
+public:
+    virtual void BuildScene() = 0;
+
+    GameObjectCollection TransferGameObjects()
+    {
+        return std::move(mGameObjects);
```

Comment:
Don't return and move, it ruins copy-elision and is often slower.

Reply:
(just return directly, it does the optimal job magically)

Reply:
So I tested this out and it comes up with some compiler errors for trying to copy a vector of `unique_ptr`s. I assumed that because this is returning a vector from a class member variable it would not implicitly move it? So it must be done explicitly.

Reply:
oh! Wow, interesting use case. cool, Today I Learned